src/charm.py

```diff
@@ -1698,51 +1678,6 @@ def _was_restore_successful(self, container: Container, service: ServiceInfo) ->
 
         return True
 
-    def _handle_processes_failures(self) -> bool:
-        """Handle Patroni and PostgreSQL OS processes failures.
-
-        Returns:
-            a bool indicating whether the charm performed any action.
-        """
-        container = self.unit.get_container("postgresql")
-
-        # Restart the Patroni process if it was killed (in that case, the PostgreSQL
-        # process is still running). This is needed until
-        # https://github.com/canonical/pebble/issues/149 is resolved.
-        if not self._patroni.member_started and self._patroni.is_database_running:
-            try:
-                container.restart(self.postgresql_service)
-                logger.info("restarted Patroni because it was not running")
-            except ChangeError:
-                logger.error("failed to restart Patroni after checking that it was not running")
-                return False
-            return True
```

Comment:
Looks practically identical to thee restart check in `_on_update_status()` itself: https://github.com/canonical/postgresql-k8s-operator/blob/6abe3749ce8b4092de4e5e8ccd8a78bf3f0e6985/src/charm.py#L1582-L1598.